File: euler_p4.py
#!/usr/bin/env python

#Largest palindrome product
#Problem # 4

import time
import pdb

t1 = time.time()

list = []

for i in range(100, 1000):
    for j in range(100, 1000):
        num = i * j
        if str(num) == str(num)[::-1]: #reverse
            list.append(num)

print(max(list))

t2 = time.time() - t1
print("runner time: %s" % t2)

File: euler_p4.rb
#!/usr/bin/env ruby

#Largest palindrome product
#Problem # 4

require 'time'

puts "starting example"
code = <<STR
arr = []

(100..999).each do |i1|
  (100..999).each do |i2|
    i3 = i1 * i2
    if i3.to_s == i3.to_s.reverse
	arr << i3
    end	 
  end
end

puts arr.max\n
STR

puts code

t1 = Time.now

arr = []

(100..999).each do |i1|
  (100..999).each do |i2|
    i3 = i1 * i2
    if i3.to_s == i3.to_s.reverse
        arr << i3
    end  
  end
end

puts arr.max

t2 = Time.now - t1
puts "runner time: #{t2}"
